src/wallet.rs

```diff
@@ -555,9 +555,34 @@ impl Wallet {
       });
     }
 
-    settings
-      .bitcoin_rpc_client(Some(name.clone()))?
-      .call::<serde_json::Value>("importdescriptors", &[serde_json::to_value(descriptors)?])?;
+    loop {
+      match settings
```

Comment:
This seems wrong, shouldn't we only ever call import descriptors once, as opposed to in a loop?